sdk/nodejs/neptune/graphGraph.ts

```diff
@@ -0,0 +1,285 @@
+// *** WARNING: this file was generated by the Pulumi Terraform Bridge (tfgen) Tool. ***
+// *** Do not edit by hand unless you're certain you know what you are doing! ***
+
+import * as pulumi from "@pulumi/pulumi";
+import * as inputs from "../types/input";
+import * as outputs from "../types/output";
+import * as enums from "../types/enums";
+import * as utilities from "../utilities";
+
+/**
+ * The AWS::NeptuneGraph::Graph resource creates an Amazon Analytics Graph.
+ *
+ * ## Example Usage
+ *
+ * ### Neptune Graph (with Vector Search configuration)
+ *
+ * Creates a Neptune Graph with 16GB provisioned memory, vector search capability with 128 dimensions, and a single replica for high availability.
+ *
+ * ```typescript
+ * import * as pulumi from "@pulumi/pulumi";
+ * import * as aws from "@pulumi/aws";
+ *
+ * // Create Neptune Graph
+ * const example = new aws.neptune.GraphGraph("example", {
+ *     graphName: "example-graph-test-20250203",
+ *     provisionedMemory: 16,
+ *     deletionProtection: false,
+ *     publicConnectivity: false,
+ *     replicaCount: 1,
+ *     kmsKeyIdentifier: "arn:aws:kms:us-east-1:123456789012:key/12345678-1234-1234-1234-123456789012",
+ *     vectorSearchConfiguration: {
+ *         vectorSearchDimension: 128,
+ *     },
+ *     tags: {
+ *         Environment: "Development",
+ *         ModifiedBy: "AWS",
+ *     },
+ * });
+ * ```
+ *
+ * ## Import
+ *
+ * Using `pulumi import`, import `aws_neptunegraph_graph` using the graph identifier. For example:
+ *
+ * ```sh
+ * $ pulumi import aws:neptune/graphGraph:GraphGraph example "graph_id"
+ * ```
+ */
+export class GraphGraph extends pulumi.CustomResource {
```

Comment:
Per https://github.com/pulumi/pulumi-aws/pull/5266#issuecomment-2703872462

We should update this to use a new module `neptunegraph`. Update the [moduleMap](https://github.com/pulumi/pulumi-aws/blob/be2becc2f7c5df1197311a23ee081bd33aabf00d/provider/resources.go#L272) and then customize the resource entry to use that module